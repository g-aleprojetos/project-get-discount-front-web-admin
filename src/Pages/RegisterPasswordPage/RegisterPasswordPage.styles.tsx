import styled from 'styled-components';
import cores from 'resources/colors';
import sizes from 'resources/sizes';
import {TextH1, TextH2} from 'Components/text';
import {Button} from 'Components/button';

export const BotaoEnviar = styled(Button).attrs({
  tipo: 'submit',
  cor: cores.CornflowerBlue,
})``;

export const ContainerDireita = styled.div`
  display: flex;
  width: 50%;
  height: 100%;
  align-items: center;
  justify-content: center;
`;

export const ContainerEsquerda = styled.div`
  display: flex;
  width: 50%;
  height: 100%;
  background-color: ${cores.mediumBlue};
`;

export const ContainerTitulo = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  width: 100%;
`;

export const Form = styled.form`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  min-width: 400px;
  padding: 20px;
  gap: ${sizes.xsmall}px;
  border-color: ${cores.black};
`;

export const Main = styled.main`
  width: 100vw;
  height: 100vh;
  position: fixed;
  display: flex;
  background-color: ${cores.lightgray};
`;

export const TextoBotao = styled(TextH2).attrs({
  cursor: 'pointer',
  cor: cores.white,
  toUpper: true,
})``;

export const TextoTitulo = styled(TextH1).attrs({
  peso: 'bold',
})``;
