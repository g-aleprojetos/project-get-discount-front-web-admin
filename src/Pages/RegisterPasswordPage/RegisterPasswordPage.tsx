import React, {useCallback, useState} from 'react';
import {z} from 'zod';
import {zodResolver} from '@hookform/resolvers/zod';
import {useForm} from 'react-hook-form';
import {Input} from 'Components/input';
import * as S from './RegisterPasswordPage.styles';

const loginSchema = z.object({
  senha: z.string().min(8),
  confirmarSenha: z.string().min(8),
});

type LoginSchema = z.infer<typeof loginSchema>;

export const RegisterPasswordPage = () => {
  const {register, handleSubmit} = useForm<LoginSchema>({
    resolver: zodResolver(loginSchema),
  });
  const [mostrarSenha, setMostrarSenha] = useState(false);

  const handleSubmitRegisterPassword = useCallback((data: LoginSchema) => {
    console.log(data);
  }, []);

  return (
    <S.Main data-testid={'registerPasswordPage'}>
      <S.ContainerEsquerda />
      <S.ContainerDireita>
        <S.Form onSubmit={handleSubmit(handleSubmitRegisterPassword)}>
          <S.ContainerTitulo>
            <S.TextoTitulo data-testid={'texto-titulo'}>
              Definir nova senha
            </S.TextoTitulo>
          </S.ContainerTitulo>
          <Input
            testId="input-nova-senha"
            label="Insira sua nova senha"
            tipo="password"
            comIcone={false}
            deveMostrarSenha={mostrarSenha}
            autoComplete="off"
            placeholder="********"
            {...register('senha')}
          />
          <Input
            testId="input-confirmar-senha"
            label="Confirme sua nova senha"
            tipo="password"
            handleMostrarSenha={setMostrarSenha}
            autoComplete="off"
            placeholder="********"
            {...register('confirmarSenha')}
          />
          <S.BotaoEnviar testId="bnt-redefinir-senha">
            <S.TextoBotao>Definir nova senha</S.TextoBotao>
          </S.BotaoEnviar>
        </S.Form>
      </S.ContainerDireita>
    </S.Main>
  );
};
