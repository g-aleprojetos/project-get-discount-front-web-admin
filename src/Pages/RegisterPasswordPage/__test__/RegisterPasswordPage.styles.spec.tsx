import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import * as S from '../RegisterPasswordPage.styles';

describe('RegisterPasswordPage.styles', () => {
  test('BotaoEnviar DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.BotaoEnviar />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('ContainerDireita DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.ContainerDireita />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('ContainerEsquerda DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.ContainerEsquerda />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('ContainerTitulo DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.ContainerTitulo />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('Form DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.Form />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('Main DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.Main />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('TextoBotao DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.TextoBotao />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('TextoTitulo DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.TextoTitulo />).toJSON();

    expect(container).toMatchSnapshot();
  });
});
