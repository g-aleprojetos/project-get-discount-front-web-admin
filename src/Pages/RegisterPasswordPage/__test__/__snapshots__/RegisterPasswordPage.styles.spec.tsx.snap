// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterPasswordPage.styles BotaoEnviar DEVE ser igual ao snapshot 1`] = `
.c0 {
  margin: 15px;
}

.c1 {
  min-width: 200px;
  background-color: #6495ED;
  border-radius: 8px;
  border: none;
  padding: 8px 60px;
  cursor: pointer;
}

<div
  className="c0"
  data-testid="bnt-pardao"
>
  <button
    className="c1"
    data-testid="bnt-pardao-button"
    type="submit"
  />
</div>
`;

exports[`RegisterPasswordPage.styles ContainerDireita DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 50%;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  className="c0"
/>
`;

exports[`RegisterPasswordPage.styles ContainerEsquerda DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 50%;
  height: 100%;
  background-color: #0000CD;
}

<div
  className="c0"
/>
`;

exports[`RegisterPasswordPage.styles ContainerTitulo DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  width: 100%;
}

<div
  className="c0"
/>
`;

exports[`RegisterPasswordPage.styles Form DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-width: 400px;
  padding: 20px;
  gap: 8px;
  border-color: #000000;
}

<form
  className="c0"
/>
`;

exports[`RegisterPasswordPage.styles Main DEVE ser igual ao snapshot 1`] = `
.c0 {
  width: 100vw;
  height: 100vh;
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #D3D3d3;
}

<main
  className="c0"
/>
`;

exports[`RegisterPasswordPage.styles TextoBotao DEVE ser igual ao snapshot 1`] = `
.c0 {
  font-size: 18px;
  font-family: AvertaStd-Regular;
  color: #FFFFFF;
  margin-top: 0px;
  text-transform: uppercase;
  line-height: 26px;
  cursor: pointer;
}

<h2
  className="c0 "
  cursor="pointer"
/>
`;

exports[`RegisterPasswordPage.styles TextoTitulo DEVE ser igual ao snapshot 1`] = `
.c0 {
  font-size: 24px;
  font-family: AvertaStd-Bold;
  color: #000000;
  margin-top: 0px;
  text-transform: none;
  line-height: 34px;
  cursor: text;
}

<h1
  className="c0 "
/>
`;
